source/common/router/config_impl.cc

```diff
@@ -1975,22 +1976,22 @@ const VirtualHostImpl* RouteMatcher::findVirtualHost(const Http::RequestHeaderMa
   // request with VHost, using wildcard match
   // Lower-case the value of the host header, as hostnames are case insensitive.
   const std::string host = absl::AsciiStrToLower(host_header_value);
-  const auto& iter = virtual_hosts_.find(host);
+  const auto iter = virtual_hosts_.find(host);
   if (iter != virtual_hosts_.end()) {
     return iter->second.get();
   }
   if (!wildcard_virtual_host_suffixes_.empty()) {
     const VirtualHostImpl* vhost = findWildcardVirtualHost(
         host, wildcard_virtual_host_suffixes_,
-        [](const std::string& h, int l) -> std::string { return h.substr(h.size() - l); });
+        [](absl::string_view h, int l) -> absl::string_view { return h.substr(h.size() - l); });
```

Comment:
Is it safe to return std::string_view here? since the string returned by the substr should be gone after lambda is executed.

Reply:
The `substr` will return a sub string view of input string view. And the input string view has same lifetime with the lower case domain.

Reply:
oh sorry, I missed that h is a absl::string_view also. I thought the h is std::string.